lib/ASTGen/CMakeLists.txt

```diff
@@ -14,116 +14,7 @@ if(SWIFT_BUILD_REGEX_PARSER_IN_COMPILER)
-  Sources/ASTGen/LegacyParse.swift
-  Sources/ASTGen/Literals.swift
-  Sources/ASTGen/ParameterClause.swift
-  Sources/ASTGen/Patterns.swift
-  Sources/ASTGen/Regex.swift
-  Sources/ASTGen/SourceFile.swift
-  Sources/ASTGen/Stmts.swift
-  Sources/ASTGen/TypeAttrs.swift
-  Sources/ASTGen/Types.swift
-
-  DEPENDENCIES
-    swiftAST
-  SWIFT_DEPENDENCIES
-    _CompilerSwiftSyntax
-    _CompilerSwiftIfConfig
-    _CompilerSwiftOperators
-    _CompilerSwiftSyntaxBuilder
-    _CompilerSwiftParser
-    _CompilerSwiftParserDiagnostics
-    _CompilerSwiftDiagnostics
-    ${ASTGen_Swift_dependencies}
-)
-
-add_pure_swift_host_library(swiftMacros STATIC
-  Sources/Macros/Macros.swift
-  Sources/Macros/PluginHost.swift
-  Sources/Macros/SourceManager.swift
-
-  DEPENDENCIES
-    swiftAST
-  SWIFT_DEPENDENCIES
-    _CompilerSwiftCompilerPluginMessageHandling
-    _CompilerSwiftDiagnostics
-    _CompilerSwiftOperators
-    _CompilerSwiftParser
-    _CompilerSwiftSyntax
-    _CompilerSwiftSyntaxMacroExpansion
-    swiftASTGen
-)
-
-add_pure_swift_host_library(swiftIDEUtilsBridging
-  Sources/SwiftIDEUtilsBridging/NameMatcherBridging.swift
-  
-  DEPENDENCIES
-    swiftAST
-  SWIFT_DEPENDENCIES
-    _CompilerSwiftIDEUtils
-    _CompilerSwiftSyntax
-    swiftASTGen
-)
-
-set(c_include_paths
-  # LLVM modules and headers.
-  "${LLVM_MAIN_INCLUDE_DIR}"
-  # Generated LLVM headers.
-  "${LLVM_INCLUDE_DIR}"
-  # Clang modules and headers.
-  ${CLANG_INCLUDE_DIRS}
-  # Bridging modules and headers.
-  "${SWIFT_MAIN_INCLUDE_DIR}"
-  # Generated C headers.
-  "${CMAKE_CURRENT_BINARY_DIR}/../../include")
-set(c_include_paths_args)
-foreach(c_include_path ${c_include_paths})
-  list(APPEND c_include_paths_args "SHELL: -Xcc -I -Xcc ${c_include_path}")
-endforeach()
-
-set(compile_options
-  ${c_include_paths_args}
-  "SHELL: -Xcc -std=c++17 -Xcc -DCOMPILED_WITH_SWIFT"
-
-  # FIXME: Needed to work around an availability issue with CxxStdlib
-  "SHELL: -Xfrontend -disable-target-os-checking"
-
-  # Necessary to avoid treating IBOutlet and IBAction as keywords
-  "SHELL:-Xcc -UIBOutlet -Xcc -UIBAction -Xcc -UIBInspectable"
-)
-
-if(CMAKE_SYSTEM_NAME STREQUAL "Windows")
-  list(APPEND compile_options
-    # Make 'offsetof()' a const value.
-    "SHELL:-Xcc -D_CRT_USE_BUILTIN_OFFSETOF"
-    # Workaround for https://github.com/swiftlang/llvm-project/issues/7172
-    "SHELL:-Xcc -Xclang -Xcc -fmodule-format=raw")
-endif()
-
-# Prior to 5.9, we have to use the experimental flag for C++ interop.
-if (CMAKE_Swift_COMPILER_VERSION VERSION_LESS 5.9)
-  list(APPEND compile_options "SHELL:-Xfrontend -enable-experimental-cxx-interop")
 else()
-  list(APPEND compile_options "-cxx-interoperability-mode=default")
-endif()
-
-if(SWIFT_BUILD_SWIFT_SYNTAX)
-  foreach(target swiftASTGen swiftIDEUtilsBridging swiftMacros)
-    target_compile_options(${target} PRIVATE ${compile_options})
-  endforeach()
+  # Dummy target for dependencies
+  add_custom_target(_CompilerRegexParser)
```

Comment:
I think this breaks situations in which `SWIFT_BUILD_REGEX_PARSER_IN_COMPILER` is false, since it is added as a dependency of `swiftASTGen`, but it is not exported, but `swiftASTGen` is. Adding it to the global `SWIFT_EXPORTS` property might be enough (or do not add it unconditionally as a `SWIFT_DEPENDENCIES` below, as it was happening before these changes).